### PR TITLE
WDP240901-13

### PR DESCRIPTION
**Pull request by dave8git**
https://projects.kodilla.com/browse/WDP240901-13

### CODE REVIEW

File: src/components/features/NewFurniture/NewFurniture.js [hunk 1]
```diff
@@ -18,12 +33,29 @@ class NewFurniture extends React.Component {
...
+      // console.log('leftAction');
+    }
+  }
 
-    const categoryProducts = products.filter(item => item.category === activeCategory);
-    const pagesCount = Math.ceil(categoryProducts.length / 8);
+  leftAction() {
+    if (this.state.activePage > 0) {
+      this.setState({ ...this.state, activePage: this.state.activePage - 1 });
+      // console.log('rightAction');
```
Comment (patrykbn) [inline src/components/features/NewFurniture/NewFurniture.js:46]: remove console.log

File: src/components/features/NewFurniture/NewFurniture.js [hunk 2]
```diff
@@ -3,13 +3,28 @@ import PropTypes from 'prop-types';
...
     activePage: 0,
     activeCategory: 'bed',
+    pagesCount: 0,
+    activePage: 0,
+    categoryProducts: [],
+    categories: [],
   };
 
+  static getDerivedStateFromProps(props, state) {
+    // return the object with new state when props changed
```
Comment (patrykbn) [inline src/components/features/NewFurniture/NewFurniture.js:19]: remove comment

File: src/components/features/NewFurniture/NewFurniture.js [hunk 3]
```diff
@@ -18,12 +33,29 @@ class NewFurniture extends React.Component {
     this.setState({ activeCategory: newCategory });
   }
 
-  render() {
-    const { categories, products } = this.props;
-    const { activeCategory, activePage } = this.state;
+  rightAction() {
+    if (this.state.activePage < this.state.pagesCount - 1) {
+      this.setState({ ...this.state, activePage: this.state.activePage + 1 });
+      // console.log('leftAction');
```
Comment (patrykbn) [inline src/components/features/NewFurniture/NewFurniture.js:39]: remove console.log

**Review by patrykbn — CHANGES_REQUESTED** [review]

File: src/components/features/NewFurniture/NewFurniture.js (hunk 1 above)
Comment (dave8git) [reply src/components/features/NewFurniture/NewFurniture.js]: console.log() usunięte.

File: src/components/features/NewFurniture/NewFurniture.js (hunk 3 above)
Comment (dave8git) [reply src/components/features/NewFurniture/NewFurniture.js]: removed.

File: src/components/features/NewFurniture/NewFurniture.js (hunk 2 above)
Comment (dave8git) [reply src/components/features/NewFurniture/NewFurniture.js]: comment removed

**Review by patrykbn — DISMISSED** [review]

**Review by patrykbn — APPROVED** [review]

**Review by matwjablonski — APPROVED** [review]

**Pull request merged**